examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -450,22 +449,36 @@ static void wfx_rsi_save_ap_info()
     }
     else
     {
-        wfx_rsi.sec.security = rsp.scan_info->security_mode;
+        wfx_rsi.sec.security = WFX_SEC_UNSPECIFIED;
         wfx_rsi.ap_chan      = rsp.scan_info->rf_channel;
         memcpy(&wfx_rsi.ap_mac.octet[0], &rsp.scan_info->bssid[0], BSSID_MAX_STR_LEN);
     }
-    if ((wfx_rsi.sec.security == RSI_WPA) || (wfx_rsi.sec.security == RSI_WPA2))
-    {
-        // saving the security before changing into mixed mode
-        security             = wfx_rsi.sec.security;
-        wfx_rsi.sec.security = RSI_WPA_WPA2_MIXED;
-    }
-    if (wfx_rsi.sec.security == SME_WPA3)
+
+    switch (rsp.scan_info->security_mode)
     {
-        // returning 3 for WPA3 when DGWIFI read security-type is called
-        security             = WPA3_SECURITY;
-        wfx_rsi.sec.security = RSI_WPA3;
+    case SME_OPEN:
+        wfx_rsi.sec.security = WFX_SEC_NONE;
+        break;
+    case SME_WPA:
+    case SME_WPA_ENTERPRISE:
+        wfx_rsi.sec.security = WFX_SEC_WPA;
+        break;
+    case SME_WPA2:
+    case SME_WPA2_ENTERPRISE:
+        wfx_rsi.sec.security = WFX_SEC_WPA2;
+        break;
+    case SME_WEP:
+        wfx_rsi.sec.security = WFX_SEC_WEP;
+        break;
+    case SME_WPA3:
+    case SME_WPA3_TRANSITION:
+        wfx_rsi.sec.security = WFX_SEC_WPA3;
+        break;
+    default:
+        wfx_rsi.sec.security = WFX_SEC_UNSPECIFIED;
```

Comment:
[SiWx917/rsi_if.c] this is a total duplication of the one from SiWx917 
factorize it !

Reply:
It will be taken up as an effort later